src/Sulu/Component/Webspace/Analyzer/RequestAnalyzer.php

```diff
@@ -97,6 +117,16 @@ public function getSegment()
         return $this->getAttribute('segment');
     }
 
+    public function changeSegment(string $segmentKey)
```

Comment:
not sure if this should be nullable. is it possible to remove the segment?

Reply:
It probably should not... If no segment is set, it should be the default segment.